src/test/codewhisperer/service/codewhisperer.test.ts

```diff
@@ -0,0 +1,97 @@
+    let telemetryEnabledDefault: boolean
+
+    beforeEach(async function () {
+        sinon.restore()
+        clientSpy = spy(await codeWhispererClient.createUserSdkClient())
+        sinon.stub(codeWhispererClient, 'createUserSdkClient').returns(Promise.resolve(clientSpy))
+        telemetryEnabledDefault = globals.telemetry.telemetryEnabled
+    })
+
+    afterEach(function () {
+        sinon.restore()
+        globals.telemetry.telemetryEnabled = telemetryEnabledDefault
+    })
+
+    it('sendTelemetryEvent for userTriggerDecision should respect telemetry optout status', async function () {
+        await sendTelemetryEventOptoutCheckHelper({
+            userTriggerDecisionEvent: {
+                sessionId: anyString(),
+                requestId: anyString(),
+                programmingLanguage: { languageName: 'python' },
+                completionType: 'BLOCK',
+                suggestionState: 'ACCEPT',
+                recommendationLatencyMilliseconds: 1,
+                timestamp: new Date(),
+            },
+        })
+    })
+
+    it('sendTelemetryEvent for codeScan should respect telemetry optout status', async function () {
+        await sendTelemetryEventOptoutCheckHelper({
+            codeScanEvent: {
+                programmingLanguage: { languageName: 'python' },
+                codeScanJobId: anyString(),
+                timestamp: new Date(),
+            },
+        })
+    })
+
+    it('sendTelemetryEvent for codePercentage should respect telemetry optout status', async function () {
+        await sendTelemetryEventOptoutCheckHelper({
+            codeCoverageEvent: {
+                programmingLanguage: { languageName: 'python' },
+                acceptedCharacterCount: 0,
+                totalCharacterCount: 1,
+                timestamp: new Date(),
+            },
+        })
+    })
+
+    it('sendTelemetryEvent for userModification should respect telemetry optout status', async function () {
+        await sendTelemetryEventOptoutCheckHelper({
+            userModificationEvent: {
+                sessionId: anyString(),
+                requestId: anyString(),
+                programmingLanguage: { languageName: 'python' },
+                modificationPercentage: 2.0,
+                timestamp: new Date(),
+            },
+        })
+    })
+
+    async function sendTelemetryEventOptoutCheckHelper(payload: TelemetryEvent) {
+        const stub = sinon.stub(clientSpy, 'sendTelemetryEvent').returns({
+            promise: () =>
+                Promise.resolve({
+                    $response: {
+                        requestId: anyString(),
+                    },
+                }),
+        } as Request<SendTelemetryEventResponse, AWSError>)
+
+        globals.telemetry.telemetryEnabled = true
+        await codeWhispererClient.sendTelemetryEvent({ telemetryEvent: payload })
+        sinon.assert.calledWith(stub, sinon.match({ optOutPreference: 'OPTIN' }))
```

Comment:
If `sendTelemetryEvent` was wrapped by the Toolkit telemetry client, then these tests could [query the telemetry batch](https://github.com/aws/aws-toolkit-vscode/blob/5e9a7555e3ef199756ae12700ea936e624f77df2/src/shared/telemetry/telemetryLogger.ts#L84) instead of checking a stubbed/mocked method.